### ISSUE TITLE
Clean up Limited API / Module State Struct code

### ISSUE THREAD

**Issue by scoder** [cython/cython#4495]
The C code to accessing globals through the module state struct uses a large number of defines. There are probably ways to do this with less code overhead.

**Comment by da-woods** [comment]
One obvious option would be to always create the `__pyx_mstate` struct, and to always access globals through that. But for the `!CYTHON_USE_MODULE_STATE` we create that as a static global rather than accessing it from the module.

**Comment by scoder** [comment]
> One obvious option would be to always create the `__pyx_mstate` struct, and to always access globals through that.

Yes, that was the original idea. I still think it should be done this way.
